File: 자료구조(스택, 큐, 해쉬, 힙)/공주 구하기.py
"""
import sys
from collections import deque
# sys.stdin = open("input.txt", 'r')

n, k = map(int, input().split())
queue = deque(range(1, n + 1))

cnt = 1
while len(queue) != 1:
    if cnt % k == 0:
        queue.popleft()
    else:
        tmp = queue.popleft()
        queue.append(tmp)
    cnt += 1
print(queue[0])
"""
import sys
from collections import deque
# sys.stdin = open("input.txt", 'r')

n, k = map(int, input().split())
queue = deque(range(1, n + 1))

while queue:
    if len(queue) == 1: break

    for _ in range(k - 1):
        tmp = queue.popleft()
        queue.append(tmp)
    queue.popleft()
print(queue[0])

File: DFS, BFS, 그래프/공주 구하기.cpp
#include <stdio.h>
#include <queue>
using namespace std;

int main(void){
	// freopen("input.txt", "r", stdin);
	int n, k;
	int i, tmp;
	scanf("%d%d", &n, &k);
	
	queue<int> q;
	for(i = 1; i <= n; i++){
		q.push(i);
	}	
	
	while(q.size() != 1){
		for(i = 1; i <= k - 1; i++){
			tmp = q.front();
			q.pop();
			q.push(tmp);
		}
		q.pop();
	}
	printf("%d", q.front());
	
	return 0;	
}
